File: Competetive programming/C/Unit 2/jolly jumper.c
#include <stdio.h>
#include <stdlib.h>

int main() {

    int n;

    printf("Enter number of elements: ");
    scanf("%d", &n);

    int arr[100];
    int diff[100] = {0};

    printf("Enter the numbers:\n");

    for(int i = 0; i < n; i++)
        scanf("%d", &arr[i]);

    for(int i = 0; i < n - 1; i++) {

        int d = abs(arr[i] - arr[i + 1]);

        if(d >= 1 && d <= n - 1)
            diff[d] = 1;
    }

    int jolly = 1;

    for(int i = 1; i <= n - 1; i++) {

        if(diff[i] == 0) {
            jolly = 0;
            break;
        }
    }

    if(jolly)
        printf("\nJolly Jumper\n");
    else
        printf("\nNot Jolly\n");

    return 0;
}

File: Competetive programming/Python/Unit 2/jolly jumper.py
print("===== JOLLY JUMPERS =====\n")

n = int(input("Enter number of elements: "))

print("\nEnter the sequence:")
print("Example: 1 4 2 3\n")

arr = list(map(int, input().split()))

diffs = set()

for i in range(n - 1):
    diffs.add(abs(arr[i] - arr[i + 1]))

jolly = True

for i in range(1, n):
    if i not in diffs:
        jolly = False
        break

if jolly:
    print("\nJolly")
else:
    print("\nNot Jolly")